cli/src/main/kotlin/model/AuthenticationStorage.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.cli.model
+
+import com.charleskorn.kaml.Yaml
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+
+import org.eclipse.apoapsis.ortserver.cli.utils.configDir
+
+private const val AUTH_FILE_NAME = "auth.json"
+
+/**
+ * Object to help with storing and retrieving authentication details.
+ *
+ * Note: Currently only single host authentication is supported.
+ */
+internal object AuthenticationStorage {
+    private val storage = mutableListOf<HostAuthenticationDetails>()
+    private val authFile = configDir.resolve(AUTH_FILE_NAME)
+
+    init {
+        if (authFile.isFile) {
+            val savedAuth = Yaml.default.decodeFromString<List<HostAuthenticationDetails>>(authFile.readText())
+
+            storage.addAll(savedAuth)
+        }
+    }
+
+    fun add(authentication: HostAuthenticationDetails) {
+        if (storage.isEmpty()) {
+            storage.add(authentication)
+        } else {
+            storage[0] = authentication
+        }
```

Comment:
Can you explain the semantic here a bit (e.g. in function documentation)? Why is it intended to overwrite a first entry, instead of prepending it, for example?

Reply:
IMO I already did that with the note in the class documentation:
```
Note: Currently only single host authentication is supported.
```

So the logic basically is:
Add the single authentication or overwrite the existing one.

Reply:
Ok, I didn't look there. Would it make sense to instead use a single (nullable) `storage` property then until this restriction is lifted? Or are you going to extend the functionality very soon to a real list?

Reply:
> Or are you going to extend the functionality very soon to a real list?

No plans to do so, yet.
Just a feature that I want at some point in time, therefore I already designed the file format in a way to support it.